### PR TITLE
[#1979] fix(test): add timeout and retry when downloading chrome

### PR DESCRIPTION
**Pull request by FANNG1**
### What changes were proposed in this pull request?
1. set connect and read timeout 30s when downloading chrome
2. add error retry 

### Why are the changes needed?
Fix: #1979 

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
could run web IT

### CODE REVIEW

**Comment by FANNG1** [comment]
set connect and read timeout to 30s, seems enough, @jerryshao @xunliu , what do you think? is it necessary to make it configurable?

**Comment by jerryshao** [comment]
What is the default value?

File: integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/utils/ChromeWebDriverProvider.java [hunk 1]
```diff
@@ -106,7 +108,14 @@ private void downloadZipFile(String url, String zipFileName, String fileName) {
     try {
       LOG.info("Download the zip file from " + url + " to " + downLoadDir);
       File chromeDriverZip = new File(ChromeWebDriverProvider.downLoadDir, zipFileName);
-      FileUtils.copyURLToFile(new URL(url), chromeDriverZip);
+
+      URL fileUrl = new URL(url);
```
Comment (yuqi1129) [inline integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/utils/ChromeWebDriverProvider.java:112]: image

`copyURLToFile` do have `conectionTimeout` and `readTimeout`, you don't need to change it.

Comment (FANNG1) [reply integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/utils/ChromeWebDriverProvider.java:112]: bingo!

**Comment by FANNG1** [comment]
> What is the default value?

seems the default is no timeout, I'll validate it.

File: integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/utils/ChromeWebDriverProvider.java [hunk 2]
```diff
@@ -103,19 +104,34 @@ private void downloadZipFile(String url, String zipFileName, String fileName) {
...
-      LOG.error(
-          "Download of: " + url + ", failed in path " + ChromeWebDriverProvider.downLoadDir, e);
-      throw new RuntimeException(e);
+    while (retryNum < 3 && Instant.now().isBefore(limit)) {
+      try {
+        LOG.info("Download the zip file from " + url + " to " + downLoadDir);
+        File chromeDriverZip = new File(ChromeWebDriverProvider.downLoadDir, zipFileName);
+        if (chromeDriverZip.exists()) {
+          chromeDriverZip.delete();
+        }
```
Comment (jerryshao) [inline integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/utils/ChromeWebDriverProvider.java:117]: Maybe we don't have to delete it when existed.

Comment (jerryshao) [reply integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/utils/ChromeWebDriverProvider.java:117]: I would suggest to rename a unique temporary name for download and unzip, when successfully downloaded and unzipped, then renaming to the target name?

Comment (FANNG1) [reply integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/utils/ChromeWebDriverProvider.java:117]: ok

Comment (FANNG1) [reply integration-test/src/test/java/com/datastrato/gravitino/integration/test/web/ui/utils/ChromeWebDriverProvider.java]: done

**Review by yuqi1129 — APPROVED** [review]
LGTM

**Pull request merged**